src/ncp/ncp_uart.cpp

```diff
@@ -176,7 +176,12 @@ void NcpUart::EncodeAndSendToUart(void)
 
     if (len > 0)
     {
-        otPlatUartSend(mUartBuffer.GetBuffer(), len);
+        if (otPlatUartSend(mUartBuffer.GetBuffer(), len) != kThreadError_None)
+        {
+            mUartBuffer.Clear();
+
+            mUartSendTask.Post();
+        }
```

Comment:
If we just post the `mUartSendTask` and not clear the `mUartBuffer` it will cause an immediate retry of the same chunk of data which I think would be better. 

To avoid back-to-back posting of the same task (thrashing) in case of error, we can consider adding a counter to keep track of number of attempts for same chunk and clear the buffer only after some reasonable number of failed attempts. It'd be even better if we can add some delay between the attempts.

Thoughts?